### PR TITLE
Evaluation successes

### PR DESCRIPTION
**Pull request by blakeandrewwood**
**Proposed changes**:

[Solves issue 1558](https://github.com/RasaHQ/rasa_nlu/issues/1558)

- Outputs successful predictions to successes.json or specified filename
- Outputs report to report.json or specified filename

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

**Comment by akelad** [comment]
Thanks for submitting this PR, we'll give it a review as soon as possible

**Comment by blakeandrewwood** [comment]
@ricwo Absolutely, thanks for the link. Will fix that asap along with the other last requests

**Comment by MetcalfeTom** [comment]
Hi @blakeandrewwood, how is this PR coming along?

**Comment by MetcalfeTom** [comment]
@ricwo unfortunately, including `output_dict` means we have to bump `scikit-learn` to `0.20` and that would [introduce a more global error](https://github.com/RasaHQ/rasa_nlu/issues/1436) - perhaps fixable, but not before the next release.

**Comment by MetcalfeTom** [comment]
Never mind, [we might be able to include `output_dict()`](https://github.com/RasaHQ/rasa_nlu/pull/1635)

File: tests/base/test_evaluation.py [hunk 1]
```diff
@@ -258,6 +261,31 @@ def test_run_cv_evaluation():
...
+                              successes_filename=None,
+                              errors_filename=None,
+                              confmat_filename=None,
+                              intent_hist_filename=None)
+
+    report = json.loads(utils.read_file(report_filename))
+
+    assert len(report.keys()) == 4
+    assert report["greet"]
+    assert result
```
Comment (tmbo) [inline tests/base/test_evaluation.py]:
```suggestion
    assert result is not None
```

File: tests/base/test_evaluation.py [hunk 2]
```diff
@@ -258,6 +261,31 @@ def test_run_cv_evaluation():
...
+                              report_filename,
+                              successes_filename=None,
+                              errors_filename=None,
+                              confmat_filename=None,
+                              intent_hist_filename=None)
+
+    report = json.loads(utils.read_file(report_filename))
+
+    assert len(report.keys()) == 4
+    assert report["greet"]
```
Comment (tmbo) [inline tests/base/test_evaluation.py]: can we assert something more specific here? I mean the value should be known right?

File: tests/base/test_evaluation.py (hunk 1 above)
Comment (tmbo) [reply tests/base/test_evaluation.py]: same here, can we assert something more specific?

**Pull request merged**